### PR TITLE
Fix Connection, Network, Server and PresenterNetwork To-Dos

### PR DESCRIPTION
**Pull request by lbgracioso**
Hey!

This PR is small and has few changes in different places. 
- Remove `connection` unused constructor
- Now `PresenterNetwork.h question` has more information about the network interface being dealt with
- Change the way `Connection::fetchInterfaces` deal with duplicates
- Improve readability on `Connection::setMAC` regex pattern and include Cisco MAC identifier
- Add specific Infiniband network MTU requirement in `Connection::setMTU` following [RFC4391](https://www.rfc-editor.org/rfc/rfc4391.html)
- Change `Network::fetchDomainName` to work with `getdomainname()` instead of `res_init`
- Add FQDN validation in `Server::setFQDN`

### CODE REVIEW

File: src/connection.cpp [hunk 1]
```diff
@@ -132,15 +119,18 @@ std::vector<std::string> Connection::fetchInterfaces()
         if (std::strcmp(ifa->ifa_name, "lo") == 0)
             continue;
 
-        interfaces.emplace_back(ifa->ifa_name);
-    }
+        bool duplicate = false;
+        for (const auto& name : interfaces) {
+            if (std::strcmp(ifa->ifa_name, name.c_str()) == 0) {
+                duplicate = true;
+                break;
```
Comment (viniciusferrao) [inline src/connection.cpp:126]: I think it should be `continue` instead of `break`.

File: src/connection.cpp [hunk 2]
```diff
@@ -132,15 +119,18 @@ std::vector<std::string> Connection::fetchInterfaces()
...
+            }
+        }
 
-    // TODO: It must have a better way to remove duplicates instead of creating
-    //       a set to filter out them. Perhaps changing the return type?
```
Comment (viniciusferrao) [inline src/connection.cpp:139]: Did we get anything good considering this TODO?

File: src/connection.cpp [hunk 3]
```diff
@@ -154,11 +144,13 @@ void Connection::setMAC(std::string_view mac)
...
 
-    // TODO: Make it easier to read and consider the Cisco MAC identifier
-    const std::regex pattern("^([0-9A-Fa-f]{2}[:-]){5}"
-                             "([0-9A-Fa-f]{2})|([0-9a-"
-                             "fA-F]{4}\\.[0-9a-fA-F]"
-                             "{4}\\.[0-9a-fA-F]{4})$");
+    const std::regex pattern(
+        R"regex(^
+    ([0-9A-Fa-f]{2}[:-]){5}        # Matches MAC address with colons or hyphens
+    ([0-9A-Fa-f]{2})|              # Or
```
Comment (viniciusferrao) [inline src/connection.cpp:150]: Didn't get this one.

File: src/connection.cpp [hunk 4]
```diff
@@ -154,11 +144,13 @@ void Connection::setMAC(std::string_view mac)
...
+    ([0-9a-fA-F]{4}\.[0-9a-fA-F]{4}\.[0-9a-fA-F]{4})  # Matches IPv6 format
```
Comment (viniciusferrao) [inline src/connection.cpp:151]: Do you have an example? IPv6?

File: src/server.cpp [hunk 5]
```diff
@@ -58,12 +58,31 @@ void Server::setHostname(std::string_view hostname)
 
 const std::string& Server::getFQDN() const noexcept { return m_fqdn; }
 
-/* TODO: Validate if FQDN is in right format */
 void Server::setFQDN(const std::string& fqdn)
 {
     if (fqdn.size() > 255)
         throw std::runtime_error("FQDN cannot be bigger than 255 characters");
 
+    const std::regex fqdnPattern(
```
Comment (viniciusferrao) [inline src/server.cpp:66]: We are missing a documentation of what the regex rule do.

File: src/connection.cpp (hunk 3 above)
Comment (lbgracioso) [reply src/connection.cpp:150]: What exactly? It's a pattern to verify if the MAC address is valid or not.

File: src/server.cpp (hunk 5 above)
Comment (lbgracioso) [reply src/server.cpp:66]: It validates whether an FQDN is valid or not.

File: src/connection.cpp (hunk 2 above)
Comment (lbgracioso) [reply src/connection.cpp:139]: I changed the way duplicate validation is done / without set... Only this

File: src/connection.cpp (hunk 1 above)
Comment (lbgracioso) [reply src/connection.cpp:126]: That's right, my bad.

File: src/connection.cpp (hunk 2 above)
Comment (viniciusferrao) [reply src/connection.cpp:139]: > I changed the way duplicate validation is done / without set... Only this

Yes but using set already do it for us and also put it in alphabetic order right? There's any advantage?

File: src/connection.cpp (hunk 3 above)
Comment (viniciusferrao) [reply src/connection.cpp:151]: > What exactly? It's a pattern to verify if the MAC address is valid or not.

I don't understand the documentation. what is the "Or"?

File: src/connection.cpp (hunk 4 above)
Comment (viniciusferrao) [reply src/connection.cpp:152]: > Matches MAC address with colons or hyphens: Example: 12:34:56:78:9A:BC
> 
> Matches IPv6 format: Example: 1a2b:3c4d:5e6f:7a8b:9c1d:1e2f:3a4b:5c6d
> 
> Matches Cisco MAC format: Example: 12ab.34cd.56ef

Alright but it makes no sense for IPv6. IPv6 is an IP address and not an MAC Address.

File: src/server.cpp (hunk 5 above)
Comment (viniciusferrao) [reply src/server.cpp:67]: > It validates whether an FQDN is valid or not.

Yes but which cases did you considered?

File: src/connection.cpp (hunk 2 above)
Comment (lbgracioso) [reply src/connection.cpp:139]: I don't think so. Just better to read (at least in my pov)

File: src/connection.cpp (hunk 3 above)
Comment (lbgracioso) [reply src/connection.cpp:151]: The "or" stands for the operator "**|**"... Before that, "Matches MAC address with colons or hyphens", and after,  "Matches IPv6 format"

Comment (viniciusferrao) [reply src/connection.cpp:151]: Can we use the regex on the same line to further enhance the reading, I was confused due to that.

File: src/server.cpp (hunk 5 above)
Comment (viniciusferrao) [reply src/server.cpp:67]: Does it allow starting with a number? We should look at the edge cases, that was I was referring to.

Comment (viniciusferrao) [reply src/server.cpp:67]: @lbgracioso can you add some unit tests for those checkers?

File: src/connection.cpp (hunk 2 above)
Comment (viniciusferrao) [reply src/connection.cpp:139]: Did you mind using set instead?

File: src/connection.cpp (hunk 3 above)
Comment (lbgracioso) [reply src/connection.cpp]: > Can we use the regex on the same line to further enhance the reading, I was confused due to that.

https://github.com/viniciusferrao/cloysterhpc/pull/27/commits/1e9882e5b5a7d1ffd4beb8593f862c4d60ed7c11
